# KPMG virtual internship
## Task 1 - Data assessment

In [1]:
# import packages
import pandas as pd
import matplotlib.pyplot as plt

In [6]:
transaction = pd.read_excel("KPMG_original.xlsx", sheet_name="Transactions", header = 1)
customerDemographic = pd.read_excel("KPMG_original.xlsx", sheet_name="CustomerDemographic", header = 1)
customerAddress = pd.read_excel("KPMG_original.xlsx", sheet_name="CustomerAddress", header = 1)

In [7]:
def check_info(df):
    print("Dataset Shape: \n", df.shape,'\n')
    print("First Five Rows: \n", df.head(),'\n')
    print(df.info(),'\n')
    print("Duplicate Check: \n", df.duplicated().sum(),'\n')
    print("Null Values: \n", df.isnull().sum(),'\n')
    print('Unique values for each column: \n', df.nunique(),'\n')
    print("Dataset Description: \n", df.describe())

In [8]:
check_info(transaction)

Dataset Shape: 
 (20000, 13) 

First Five Rows: 
    transaction_id  product_id  customer_id transaction_date  online_order   
0               1           2         2950       2017-02-25           0.0  \
1               2           3         3120       2017-05-21           1.0   
2               3          37          402       2017-10-16           0.0   
3               4          88         3135       2017-08-31           0.0   
4               5          78          787       2017-10-01           1.0   

  order_status           brand product_line product_class product_size   
0     Approved           Solex     Standard        medium       medium  \
1     Approved   Trek Bicycles     Standard        medium        large   
2     Approved      OHM Cycles     Standard           low       medium   
3     Approved  Norco Bicycles     Standard        medium       medium   
4     Approved  Giant Bicycles     Standard        medium        large   

   list_price  standard_cost  product_firs

There are some missing values in brand, product_line, product_class, product_size, standard_cost and product_first_sold_date. Meaning certain product data are missing. As the missing value proportion is small, it is recommended to remove these records for better analysing. Online_order column has 360 missing values, will leave it as is.  
Number of unique customer_id: 3494  

In [9]:
check_info(customerAddress)

Dataset Shape: 
 (3999, 6) 

First Five Rows: 
    customer_id              address  postcode            state    country   
0            1   060 Morning Avenue      2016  New South Wales  Australia  \
1            2  6 Meadow Vale Court      2153  New South Wales  Australia   
2            4   0 Holy Cross Court      4211              QLD  Australia   
3            5  17979 Del Mar Point      2448  New South Wales  Australia   
4            6     9 Oakridge Court      3216              VIC  Australia   

   property_valuation  
0                  10  
1                  10  
2                   9  
3                   4  
4                   9   

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3999 entries, 0 to 3998
Data columns (total 6 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   customer_id         3999 non-null   int64 
 1   address             3999 non-null   object
 2   postcode            3999 non-null   int6

No missing values in this dataset.  
Number of unique customer_id: 3999  
There are 5 unique state record, with a further look through, "New South Wales" and "NSW" were both written for "NSW", as other states all use abbreviations. "New South Wales" will need to be changed into "NSW"  


In [10]:
check_info(customerDemographic)

Dataset Shape: 
 (4000, 13) 

First Five Rows: 
    customer_id      first_name  last_name  gender   
0            1         Laraine  Medendorp       F  \
1            2             Eli    Bockman    Male   
2            3           Arlin     Dearle    Male   
3            4          Talbot        NaN    Male   
4            5  Sheila-kathryn     Calton  Female   

   past_3_years_bike_related_purchases                  DOB   
0                                   93  1953-10-12 00:00:00  \
1                                   81  1980-12-16 00:00:00   
2                                   61  1954-01-20 00:00:00   
3                                   33  1961-10-03 00:00:00   
4                                   56  1977-05-13 00:00:00   

                job_title job_industry_category     wealth_segment   
0     Executive Secretary                Health      Mass Customer  \
1  Administrative Officer    Financial Services      Mass Customer   
2      Recruiting Manager              Prop

125 missing values in last_name, but no empty values in first_name, so will leave this. Missing job_title and job_industry_category might cause some problem when modeling customer behaviour patterns based on job. The "default" column doesn't contains any meaningful information, may need to be deleted.  
Number of unique customer_id: 4000  
There are 6 unique genders, but when taking a further look, Female are written as "F"/"Femal"/"Female, and Male are written as "M"/"Male", these need to be modified to standard "Female" and "Male". Additionally, unique value is "U" need to be clarified with.  
DOB is not a date datatype, need to be modified and do a further check


In [13]:
customerDemographic['DOB'] = pd.to_datetime(customerDemographic['DOB'])

In [14]:
customerDemographic.describe()

customer_id  past_3_years_bike_related_purchases   
count  4000.000000                          4000.000000  \
mean   2000.500000                            48.890000   
min       1.000000                             0.000000   
25%    1000.750000                            24.000000   
50%    2000.500000                            48.000000   
75%    3000.250000                            73.000000   
max    4000.000000                            99.000000   
std    1154.844867                            28.715005   

                              DOB       tenure  
count                        3913  3913.000000  
mean   1977-07-12 23:56:41.277792    10.657041  
min           1843-12-21 00:00:00     1.000000  
25%           1968-01-22 00:00:00     6.000000  
50%           1977-07-22 00:00:00    11.000000  
75%           1987-02-28 00:00:00    15.000000  
max           2002-03-11 00:00:00    22.000000  
std                           NaN     5.660146

The DOB ranged from 1843-12-21 to 2022-03-11, the earliest date of DOB is not right. If it's just single record, this record need to be removed.